### PR TITLE
Changed the test for symmetry of a Number.

### PR DESCRIPTION
**Pull request by mcognetta**
Changed the symmetry test for Numbers. Previously it always returned true. Now it tests whether the argument passed in is equal to itself. This causes `issymmetric(NaN)` to return false when previously it returned true.

### CODE REVIEW

**Comment by andreasnoack** [comment]
Could you add a test for this behavior?

File: base/linalg/generic.jl [hunk 1]
```diff
@@ -712,7 +712,7 @@ function issymmetric(A::AbstractMatrix)
     return true
 end
 
-issymmetric(x::Number) = true
+issymmetric(x::Number) = x == x
```
Comment (nalimilan) [inline base/linalg/generic.jl]: Probably more explicit as `!isnan(x)`?

Comment (andreasnoack) [inline base/linalg/generic.jl]: I thought about that but I actually like `x==x` because it's the definition of symmetry collapsed to a scalar.

**Comment by mcognetta** [comment]
Yes, I will add tests to this and my other pr tomorrow.

**Pull request merged**